File: app/src/test/java/com/example/adiphong/rx2/FromActionTest.java
package com.example.adiphong.rx2;

import org.junit.Test;

import io.reactivex.Completable;
import io.reactivex.Maybe;

/**
 * Created by adiphong on 27/11/2017 AD.
 */

public class FromActionTest {

    @Test
    public void maybeFromAction() {
        Maybe.fromAction(() -> System.out.println("-------- from action"))
                .subscribe(
                        s -> System.out.println("------- maybeFromAction onSuccess " + s),
                        throwable -> System.out.println("---------- maybeFromAction error"),
                        () -> System.out.println("-------------- maybeFromAction onComplete ")
                );
    }

    @Test
    public void completableFromAction() {
        // ignore result
        Completable.fromCallable(
                () -> "hello")
                .subscribe(
                        () -> System.out.println("-------------- completableFromAction onComplete"),
                        throwable -> System.out.println("-------- completableFromAction onError")
                );
    }
}

File: app/src/test/java/com/example/adiphong/rx2/FromActionTest.kt
package com.example.adiphong.rx2

import io.reactivex.Completable
import io.reactivex.Maybe
import org.junit.Test

/**
 * Created by adiphong on 27/11/2017 AD.
 */

class FromActionTest {

    @Test
    fun maybeFromAction() {
        Maybe.fromAction<Any> { println("-------- from action") }
                .subscribe(
                        { s -> println("------- maybeFromAction onSuccess $s") },
                        { throwable -> println("---------- maybeFromAction error") },
                        { println("-------------- maybeFromAction onComplete ") }
                )
    }

    @Test
    fun completableFromAction() {
        // ignore result
        Completable.fromCallable { "hello" }
                .subscribe(
                        { println("-------------- completableFromAction onComplete") },
                        { throwable -> println("-------- completableFromAction onError") }
                )
    }
}
